### PR TITLE
HPCC-32683 Fix issues with postmortem and container death

### PR DESCRIPTION
**Pull request by jakesmith**
helm changes:
- Consistently generate command via addCommandAndLifecycle, and collect added container information in lifeCycleCtx based on containers added to lifeCycleCtx
- Add terminationGracePeriodSeconds option
- Mount ephemeral directories in distinct subPaths.
- Fix issues with postmortem's from different containers overwriting one another
- Add addPostRunContainer to generate postrun container, to monitor running containers.

script changes:
- add container_watch.sh for postrun pod.
- move all options handling into check_executes.sh

code changes:
- code grace time into k8s job file into, so that postjob clearup can also use
- Clear wuid file (prevents spurious error association)

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
Mark I would value you reviewing this change as well.

**Review by ghalliday — COMMENTED** [review]
Looks good as far as I can tell.  I didn't see any issues.

File: helm/hpcc/templates/thor.yaml [hunk 1]
```diff
@@ -234,31 +241,36 @@ data:
         spec:
           {{- include "hpcc.placementsByJobTargetType" (dict "root" .root "job" $thorWorkerJobName "target" .me.name "type" "thor") | indent 10 }}
           serviceAccountName: hpcc-default
+          terminationGracePeriodSeconds: {{ .terminationGracePeriodSeconds | default 60 }}
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply helm/hpcc/templates/thor.yaml]: yes, will change it to 600.

File: initfiles/bin/container_watch.sh [hunk 2]
```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply initfiles/bin/container_watch.sh:1]: Have added comment blocks.

**Comment by jakesmith** [comment]
@ghalliday - see 2nd commit.

**Review by ghalliday — APPROVED** [review]

**Review by mckellyln — DISMISSED** [review]
Looks good to me.
Approved.
(But I am not an expert in this area)

**Review by mckellyln — APPROVED** [review]
Looks good to me.
Approved.
(But I am not an expert in the area)

**Comment by jakesmith** [comment]
@ghalliday - now squashed.

**Pull request merged**